caim_base/views/fosterer_profile.py

```diff
@@ -513,11 +527,39 @@ def edit(request, stage_id):
                 formsets_are_valid = False
 
         if stage_id == "household-details":
+            # Number of cohabitant fields filled must be equal, or great then the listed number of people in the
+            # home.
+            num_people_in_home = int(person_in_home_detail_formset.data['num_people_in_home'])
+            if num_people_in_home:
+                for index, person_in_home_detail_form in enumerate(person_in_home_detail_formset):
+                    if index <= num_people_in_home - 1:
+                        person_in_home_detail_form.full_clean()
+                        if not any(person_in_home_detail_form.cleaned_data.values()):
+                            formsets_are_valid = False
+                            messages.error(
+                                request,
+                                f"Ensure the number of people in your home ({num_people_in_home}) matches " \
+                                f"the number of \"Person in Home Details\" fields you have filled out."
+                            )
+
+            # Ensure if a user has selected `Rent` they must fill out rent details.
+            rent_own = person_in_home_detail_formset.data['rent_own']
+            if rent_own == FostererProfile.RentOwn.RENT:
+                if not person_in_home_detail_formset.data['rent_restrictions']:
+                    formsets_are_valid = False
+                    messages.error(request, 'Please describe any pet restrictions that are in place.')
+                if not person_in_home_detail_formset.data['landlord_contact_text']:
+                    formsets_are_valid = False
+                    messages.error(request, 'Please provide your landlord’s contact information below.')
+
             if not person_in_home_detail_formset.is_valid():
                 formsets_are_valid = False
 
         form_is_valid = form.is_valid()
 
+        if not formsets_are_valid:
+            messages.error(request, "Form is not valid due to errors.")
+
         if form_is_valid and formsets_are_valid:
```

Comment:
Since we're checking `if not formsets_are_valid` above now, it looks like we can ditch the second condition here